src/main/java/org/opentripplanner/middleware/otp/core/util/IntUtils.java

```diff
@@ -0,0 +1,46 @@
+package org.opentripplanner.middleware.otp.core.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IntUtils {
+    private static final Logger LOG = LoggerFactory.getLogger(IntUtils.class);
+
+    public static int getIntFromString(String input) {
+        int retVal = 0;
+
+        Integer n = getIntegerFromString(input);
+        if (n != null) {
+            retVal = n.intValue();
+        } else if (input != null) {
+            retVal = input.hashCode();
+        }
+        return retVal;
+    }
+
+    public static Integer getIntegerFromString(String input) {
+        try {
+            return Integer.parseInt(input);
+        } catch (Exception e) {
+            try {
+                String in = input.replaceAll("\\D", "");
```

Comment:
regex should have comments.

Reply:
Will update if we are keeping OTP classes